### ISSUE TITLE
Introduce testing with Azure

### ISSUE THREAD

**Issue by julienschat** [hmgaudecker/econ-project-templates#52]

**Comment by julienschat** [comment]
Added an untested .travis.yml file (d0bcf01c8ef2ceab177c865469d6504e9cf68ac8). How can other contributors set up travis and test it?

**Comment by hmgaudecker** [comment]
Seems to be all set, thanks! 

However, it starts running, but creation of the conda environment fails, likely because of a space left there by cookiecutter (python= 3.7 instead of python=3.7):

```
$ conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION jupyterlab=0.32 pandas=0.23 python= 3.7 python-graphviz=0.8
Collecting package metadata: ...working... done
Solving environment: ...working... failed
PackagesNotFoundError: The following packages are not available from current channels:
  - 3.7

```

**Comment by hmgaudecker** [comment]
Btw, it seems like we can get rid of the virtualenv stuff now that we have conda included there, can't we?

**Comment by julienschat** [comment]
I fixed the small bug 7c47c6a594edb59c9eb91be1e8c28c1a0d33c047. Which virtualenv stuff do you mean?

**Comment by hmgaudecker** [comment]
Travis uses virtualenv and I thought that some point we had more on that, but that does not seem the case anymore.

A couple of things:

- Installation of miniconda should be sufficient, no? Environments will be created as part of the testing
- Along the same lines: Do we still need requirements.txt and tox.ini for anything?
- It currently errors out, seems pretty unrelated to our own code though. Funny. https://travis-ci.org/hmgaudecker/econ-project-templates

**Comment by hmgaudecker** [comment]
Experts to be found here https://github.com/orgs/OpenSourceEconomics/people

**Comment by hmgaudecker** [comment]
3.7 errors because of https://github.com/travis-ci/travis-ci/issues/9815, we should just set the Python version via a CONDA_PYTHON. 

3.6 fails because we attempt to activate the environment, maybe we should try without that and install everything into the base environment. Some stuff to get that to work here: https://github.com/conda/conda/issues/8462

As of today, the conda docs on the issue seem outdated (still have the pre-4.6 "source activate" in there): https://conda.io/projects/conda/en/latest/user-guide/tasks/use-conda-with-travis-ci.html

**Comment by hmgaudecker** [comment]
Update: After 6e18fb85c64275, both 3.6 and 3.7 fail because the conda environment cannot be found.

**Comment by hmgaudecker** [comment]
Azure now does Linux, Mac and Windows all in one, so let's use that. 

I mostly set this up now, however, 

- `.azure-pipelines.yaml` is still exactly the version found at https://github.com/tobiasraabe/cookiecutter-research-template and needs adjustment (no tox, ... )
- Also, it would be great to add badges as here: https://github.com/tobiasraabe/cookiecutter-research-template/blob/master/README.rst, the relevant portions of the Azure site for this repo should be https://dev.azure.com/hmgaudecker/econ-project-templates

We do not need to support Python 3.6, I think.

**Comment by hmgaudecker** [comment]
Please also check changes to the yaml file and discussion at https://github.com/iza-institute-of-labor-economics/gettsim/pull/1

**Comment by raholler** [comment]
Sorry for the noob question: What is azure supposed to test in this project? Only the pytests? Or also whether the documentation can be compiled? And anything else?

**Comment by raholler** [comment]
should we add an environment file for developers?

**Comment by hmgaudecker** [comment]
As discussed -- yes, please!

**Comment by raholler** [comment]
I will completely leave out windows for now, the pytests will also fail, as long as environments cannot be activated in powershell.

**Comment by hmgaudecker** [comment]
Fixed in 27d24fa
